src/ctia/entity/investigation.clj

```diff
@@ -1,60 +1,38 @@
 (ns ctia.entity.investigation
-  (:require [ctia.domain.entities :refer [default-realize-fn]]
-            [ctia.store :refer :all]
-            [ctia.http.routes
-             [common :refer [BaseEntityFilterParams PagingParams SourcableEntityFilterParams]]
-             [crud :refer [entity-crud-routes]]]
-            [ctia.schemas
-             [utils :as csu]
-             [core :refer [def-stored-schema
-                           CTIAEntity]]
-             [sorting :as sorting]]
-            [ctia.schemas.graphql
-             [sorting :as graphql-sorting]
-             [flanders :as flanders]
-             [helpers :as g]
-             [pagination :as pagination]]
-            [ctia.stores.es
-             [mapping :as em]
-             [store :refer [def-es-store]]]
-            [ctim.schemas.investigation :as inv]
-            [flanders
-             [schema :as f-schema]
-             [spec :as f-spec]
-             [utils :as fu]]
-            [schema-tools.core :as st]
-            [schema.core :as s]
-            [ctia.schemas.graphql.ownership :as go]))
-
-(s/defschema Investigation
-  (st/merge (f-schema/->schema inv/Investigation)
-            CTIAEntity
-            {s/Keyword s/Any}))
-
-(f-spec/->spec inv/Investigation "investigation")
-
-(s/defschema PartialInvestigation
-  (st/merge (f-schema/->schema (fu/optionalize-all inv/Investigation))
-            CTIAEntity
-            {s/Keyword s/Any}))
-
-(s/defschema PartialInvestigationList
-  [PartialInvestigation])
-
-(s/defschema NewInvestigation
-  (st/merge (f-schema/->schema inv/NewInvestigation)
-            CTIAEntity
-            {s/Keyword s/Any}))
-
-(f-spec/->spec inv/NewInvestigation "new-investigation")
-
-(def-stored-schema StoredInvestigation Investigation)
-
-(s/defschema PartialStoredInvestigation
-  (csu/optional-keys-schema StoredInvestigation))
-
-(def realize-investigation
-  (default-realize-fn "investigation" NewInvestigation StoredInvestigation))
+  (:require
+   [ctia.http.routes
+    [common :refer [BaseEntityFilterParams
+                    PagingParams
+                    SourcableEntityFilterParams]]
+    [crud :refer [entity-crud-routes]]]
+   [ctia.stores.es
+    [mapping :as em]
+    [store :refer [def-es-store]]]
+   [ctia.entity.investigation.schemas
+    :refer [Investigation
+            PartialInvestigation
+            PartialStoredInvestigation
+            NewInvestigation
+            StoredInvestigation
+            PartialInvestigationList
+            realize-investigation]]
+   [schema-tools.core :as st]
+   [schema.core :as s]
+   [ctia.schemas.sorting :as sorting]))
+
+(def snapshot-action-fields-mapping
+  {:object_ids {:type "text"
```

Comment:
I would use `token` here, unless you have special ids and need to search in sub tokens.

Reply:
Do you mean `token` instead of `text`?

Reply:
yes